File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        int n=arr.size();
        vector<int> pre(arr.size(),0);
        int x=0,ans=0;
        for(int i=0;i<arr.size();++i)
        {
            pre[i]=x;
            x^=arr[i];
        }
        for(int i=0;i<n;++i)
        {
            for(int j=i+1;j<n;++j)
            {
                for(int k=j;k<n;++k)
                {
                    int l=pre[j]^pre[i];
                    int r=pre[k]^pre[j]^arr[k];
                    if(l==r)
                    ans++;
                }
            }
        }
        return ans;
    }
};

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        n = len(arr)
        pre = [0] * len(arr)
        x = 0
        ans = 0
        for i in range(n):
            pre[i] = x
            x ^= arr[i]
        for i in range(n):
            for j in range(i+1, n):
                for k in range(j, n):
                    l = pre[j] ^ pre[i]
                    r = pre[k] ^ pre[j] ^ arr[k]
                    if l == r:
                        ans += 1
        return ans

        